ipycytoscape/cytoscape.py

```diff
@@ -376,15 +376,15 @@ def add_graph_from_networkx(self, g, directed=False):
             node_instance = Node()
             _set_attributes(node_instance, data)
             if 'id' not in data:
-                node_instance.data['id'] = node
+                node_instance.data['id'] = str(node)
             node_list.append(node_instance)
         self.add_nodes(node_list)
 
         edge_list = list()
         for source, target, data in g.edges(data=True):
             edge_instance = Edge()
-            edge_instance.data['source'] = source
-            edge_instance.data['target'] = target
+            edge_instance.data['source'] = str(source)
+            edge_instance.data['target'] = str(target)
```

Comment:
Similarly here should check if it subclasses `ipycytoscape.Node`.
```suggestion
            edge_instance.data['source'] = source.data['id'] if issubclass(type(source), Node) else str(source)
            edge_instance.data['target'] = target.data['id'] if issubclass(type(target), Node) else str(target)
```

Reply:
@jaydonnell I think implementing this suggestion will help with the problems you are seeing with the edges.